src/runtime_src/tools/xclbinutil/Section.cxx

```diff
@@ -94,59 +94,47 @@ Section::getSupportedKinds()
 }
 
 
-// Next phase: This method will be refactor such that the underlying 
-// section class will initialize its own section data.
 void
-Section::registerSectionCtor(enum axlf_section_kind eKind,
-                             const std::string& cmdName,
-                             const std::string& nodeName,
-                             bool bSupportsSubSections,
-                             bool bSupportsIndexing,
-                             Section_factory sectionCtor) {
-
+Section::addSectionType(std::unique_ptr<SectionInfo> sectionInfo)
+{
   // Some error checking
-  if (cmdName.empty()) {
-    auto errMsg = boost::format("ERROR: CMD name for the section kind (%d) is empty. This needs to be defined.") % eKind;
+  if (sectionInfo->name.empty()) {
+    auto errMsg = boost::format("ERROR: CMD name for the section kind (%d) is empty. This needs to be defined.") % sectionInfo->eKind;
     throw std::runtime_error(errMsg.str());
   }
 
   // Get the collection of sections
   auto & sections = getSectionTypes();
```

Comment:
Great question and there are several reasons for this:
1. Static variable initialization.  Being that the init() method is being executed as part of static variable initialial variable, there is no guarantee that the SectionInfo container is available.   This is because there is no dependency between the various static variables.   Yes, this code is being executed BEFORE main is called.
2. To force the initialization order dependency, you place the static variables in a static method.
3. I encapsulates access to the static collection.